### PR TITLE
WIP: Kc sig fapi#58

### PR DESCRIPTION
**Pull request by akoserwal**
issue: https://github.com/keycloak/kc-sig-fapi/issues/58

### CODE REVIEW

**Review by tnorimat — COMMENTED** [review]
Reviewing my [PR treating client settings](https://github.com/keycloak/keycloak/pull/5083/files). It seems that we need not to modify `ClientRepresentation` and `ClientModel`.  LGTM for other 4 files.

**Comment by akoserwal** [comment]
@tnorimat: please take a look.

Screenshot 2020-12-01 at 1 37 54 PM

**Comment by tnorimat** [comment]
@akoserwal Hello. As for your commits, LGTM. Thank you.

Additionally, IMO, we need to do the following 2 works.
- If CIBA parameters are set on the client, use them instead of ones set on the realm (namely, CIBA Policy)
- Write arquillian integration test

CIBA parameters are used in `BackchannelAuthenticationEndpoint`, `DelegateDecoupledAuthenticationProvider` and `TokenEndpoint`. These classes look up CIBA parameters set on CIBA Policy. If CIBA parameters are set on the client, we need to use them.

As for the arquillian integration test, you can add the test methods to the existing `CIBATest`. It might be helpful to look at `CIBATest.testCIBAPolicy()`.

File: services/src/main/java/org/keycloak/protocol/ciba/endpoints/BackchannelAuthenticationEndpoint.java [hunk 1]
```diff
@@ -124,12 +136,13 @@ public Response processGrantRequest() {
...
-        provider.doBackchannelAuthentication(client, request, policy.getExpiresIn(), authResultId, userSessionIdWillBeCreated);
+        provider.doBackchannelAuthentication(client, request, expireIn, authResultId, userSessionIdWillBeCreated);
 
         ObjectNode response = JsonSerialization.createObjectNode();
         response.put(CIBAConstants.AUTH_REQ_ID, authReqId);
-        response.put(CIBAConstants.EXPIRES_IN, policy.getExpiresIn());
-        if (policy.getInterval() > 0) response.put(CIBAConstants.INTERVAL, policy.getInterval());
+
+        response.put(CIBAConstants.EXPIRES_IN, expireIn);
+        if (policy.getInterval() > 0) response.put(CIBAConstants.INTERVAL, interval);
```
Comment (tnorimat) [inline services/src/main/java/org/keycloak/protocol/ciba/endpoints/BackchannelAuthenticationEndpoint.java:145]: IMO, how about the following?
```
        if (interval > 0) response.put(CIBAConstants.INTERVAL, interval);
```

**Review by tnorimat — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/keycloak/protocol/ciba/endpoints/BackchannelAuthenticationEndpoint.java (hunk 1 above)
Comment (akoserwal) [reply services/src/main/java/org/keycloak/protocol/ciba/endpoints/BackchannelAuthenticationEndpoint.java:145]: yes, right. I missed it. I'll fix it. thank you

**Comment by tnorimat** [comment]
@akoserwal  Thank you!

**Pull request merged**